### PR TITLE
service binding discoverability labels in dev catalog

### PR DESCRIPTION
**Pull request by christianvogt**
POC for https://issues.redhat.com/browse/ODC-6467

Enhance the dev catalog extensions to allow for contributing extra metadata to other catalog item contributions.

Operator backed services are contributed to the dev catalog by OLM.
Service binding will eventually become it's own plugin.

Created a new extension `console.catalog/item-metadata` which would allow SB to enhance the CSV catalog items by adding a `bindable` label. 

New metadata extensions allow for addition of `tags`, `attributes`, and `labels` to catalog items.

Added labels to quick search.

**TODO**
- Confirm labels and filters with UX.
- Sort filters alphabetically
- Use watcher instead of k8sGet

**Screen shots / Gifs for design review**: 

image

image

image

**Unit test coverage report**: 

**Test setup:**

**Browser conformance**: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [x] Edge

cc @serenamarie125

### CODE REVIEW

File: frontend/packages/dev-console/src/components/catalog/providers/useBindableItemMetadataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+} from '@console/dynamic-plugin-sdk';
+import { fetchBindableServices } from '../../topology/bindable-services/fetch-bindable-services-utils';
+
+const useBindableItemMetadataProvider: ExtensionHook<CatalogItemMetadataProviderFunction> = () => {
+  const [bindableServices, setBindableServices] = React.useState([]);
+  const [loaded, setLoaded] = React.useState(false);
+  const [error, setError] = React.useState(null);
+
+  React.useEffect(() => {
+    fetchBindableServices()
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/catalog/providers/useBindableItemMetadataProvider.ts:16]: TODO use a watcher instead of `k8sGet`

**Comment by divyanshiGupta** [comment]
@beaumorley can you PTAL at this PR and approve if the labels and filters looks good to you?

**Comment by divyanshiGupta** [comment]
@christianvogt can you add a few unit tests and fix the ones that are failing? Also please remove WIP.

**Comment by divyanshiGupta** [comment]
Created a followup story https://issues.redhat.com/browse/ODC-6491 for the remaining todos.

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]
@christianvogt code looks good to me and I also ran it locally and it works as expected. Apart from that I think there is 1 issue that is when there are no bindable services the filters on the sidepanel are not shown
Screenshot 2022-02-25 at 4 00 29 PM
IMO the filters should be shown so that the user knows that there are no bindable services at the moment.

cc: @beaumorley 

WDYT?

**Comment by divyanshiGupta** [comment]
/retitle service binding discoverability labels in dev catalog

**Comment by christianvogt** [comment]
@divyanshiGupta the filter mechanism is same across all catalog pages. ie if all items match a single filter then that filter is not shown.

If @beaumorley wants to change this, I suggest creating a separate issue to track this.

**Comment by divyanshiGupta** [comment]
> @divyanshiGupta the filter mechanism is same across all catalog pages. ie if all items match a single filter then that filter is not shown.
> 
> If @beaumorley wants to change this, I suggest creating a separate issue to track this. But this was the decision made by UX at that time of implementing filters.

@christianvogt Thanks for clarifying. Will create another issue if required.

**Comment by christianvogt** [comment]
Addressed comments and added tests.
Remainder of TODOs to be completed as followup work.

**Review by divyanshiGupta — APPROVED** [review]
Tested the PR and it works as expected.
/lgtm

**Pull request merged**